chia/util/config.py

```diff
@@ -41,35 +47,72 @@ def config_path_for_filename(root_path: Path, filename: Union[str, Path]) -> Pat
     return root_path / "config" / filename
 
 
+def get_config_lock(root_path: Path, filename: Union[str, Path]) -> BaseFileLock:
+    lock_path: Path = config_path_for_filename(root_path, filename).with_suffix(".lock")
+    if not lock_path.exists():
+        lock_path.touch(exist_ok=True)
+    return FileLock(lock_path)
+
+
 def save_config(root_path: Path, filename: Union[str, Path], config_data: Any):
+    # This must be called under an acquired config lock
```

Comment:
Can we require that save_config() has a 'lock' object of some sort passed-in to prove that the lock was acquired?